agir/events/components/eventPage/EventPage.js

```diff
@@ -373,9 +374,28 @@ export const ConnectedEventPage = (props) => {
   const isSessionLoaded = useSelector(getIsSessionLoaded);
   const dispatch = useDispatch();
 
-  const { data: eventData } = useSWR(
-    api.getEventEndpoint("getEvent", { eventPk })
+  const fetcher = async (url) => {
+    const res = await fetch(url);
+    if (!res.ok) {
+      let error = new Error("An error occurred while fetching the data.");
+      error.status = res.status;
+      throw error;
+    }
+    return res.json();
+  };
+
+  const { data: eventData, error } = useSWR(
+    api.getEventEndpoint("getEvent", {
+      eventPk,
+    }),
+    fetcher,
+    {
+      onErrorRetry: (error) => {
+        if ([404, 410, 403].includes(error.status)) return;
+      },
+    }
```

Comment:
Je crois qu'il n'y a pas besoin de surcharger le `fetcher` par défaut. S'il y a une erreur tu peux le récuperer directement dans le retour de `useSWR` :
```suggestion
  const { data: eventData, error } = useSWR(
    api.getEventEndpoint("getEvent", {
      eventPk,
    }),
    {
      onErrorRetry: (error) => {
        if ([404, 410, 403].includes(error.status)) return;
      },
    }
```

Reply:
D'ailleurs, en règle général, c'est mieux d'éviter de définir des fonctions statiques directement dans la fonction du composant car elle sera rédefinie à chaque render : si on n'a pas besoin d’accéder au scope du composant on définit la fonction à l'extérieur, si on a besoin du scope du composant on utilise le hook `useCallback`.